src/main/java/com/recursive_pineapple/nuclear_horizons/reactors/tile/TileReactorCore.java

```diff
@@ -1040,12 +1043,32 @@ public int addAirHeat(int airHeat) {
                 this.coolantTank.getFluidAmount(),
                 this.hotCoolantTank.getCapacity() - this.hotCoolantTank.getFluidAmount());
 
-            int consumedCoolant = Math.min(roundedHeat / coolantCache.specificHeatCapacity, heatableCoolant);
+            int consumedCoolant;
+            //BWR
+            if(this.coolantCache.cold.getName().equals("distilled_water")) {
+                consumedCoolant = Math.min(
+                    roundedHeat / (coolantCache.specificHeatCapacity),
+                    Math.min(
+                        this.coolantTank.getFluidAmount(),
+                        (this.hotCoolantTank.getCapacity() - this.hotCoolantTank.getFluidAmount())/Config.BWR_STEAM_PER_HU_MULTIPLIER
+                    ));
+            }
+            //conventional coolants
+            else {
+                consumedCoolant = Math.min(roundedHeat / coolantCache.specificHeatCapacity, heatableCoolant);
+            }
             this.roundedHeat -= consumedCoolant * coolantCache.specificHeatCapacity;
             this.addedHeat += consumedCoolant * coolantCache.specificHeatCapacity;
 
-            this.coolantTank.drain(consumedCoolant, true);
-            this.hotCoolantTank.fill(new FluidStack(coolantCache.hot, consumedCoolant), true);
+            //for BWRs, convert distilled water to a configured amount of steam instead of the same quantity of hot coolant
```

Comment:
It might be useful to make this fluid multiplying logic generic, but I can't think of anything else that would use it so this is fine for now.

Reply:
I considered adding generic values but I didn't want to significantly change any templates for an exceptional case